### PR TITLE
clock fixes and changes to undef all macros which librpmi headers are defining

### PR DESCRIPTION
**Pull request by pathakraul**

### CODE REVIEW

File: lib/librpmi_internal.h [hunk 1]
```diff
@@ -34,20 +34,33 @@
 #define offsetof(TYPE, MEMBER) ((rpmi_size_t) &((TYPE *)0)->MEMBER)
 #endif
 
+#undef container_of
```
Comment (avpatel) [inline lib/librpmi_internal.h:37]: Instead of `#undef` use `#ifndef` here and everywhere below.

Comment (pathakraul) [reply lib/librpmi_internal.h:37]: We should not rely on host firmware for these because we will have no control over how these macros will be implemented by host fw

Comment (avpatel) [reply lib/librpmi_internal.h:37]: in which case, add "rpmi_" prefix to these macro names and drop the `#undef`

Comment (pathakraul) [reply lib/librpmi_internal.h:37]: some macros here may have common understanding but others like CLAMP, ROUNDUP, ROUNDDOWN, there is none

Comment (avpatel) [reply lib/librpmi_internal.h:37]: adding "rpmi_" prefix does'nt change the understanding of macros.

Comment (pathakraul) [reply lib/librpmi_internal.h:37]: that comment was before yours updated :)

Comment (pathakraul) [reply lib/librpmi_internal.h]: done

File: include/librpmi.h [hunk 2]
```diff
@@ -378,6 +378,7 @@ void rpmi_shmem_destroy(struct rpmi_shmem *shmem);
  * a built-in service group of the RPMI context and is always available.
  */
 struct rpmi_context;
+struct rpmi_transport;
```
Comment (avpatel) [inline include/librpmi.h:381]: Why this change ?

Comment (pathakraul) [reply include/librpmi.h:381]: Due to movement of the groups in header, this structure is gone below where it was referred. So instead of moving that group again above i declared here. Not sure why it didnt failed when i build with previous PR but it failed now so made the change.

Comment (avpatel) [reply include/librpmi.h:381]: If this is to fix compile warning then move LIBRPMI_TRANSPORT_INTERFACE  before LIBRPMI_CONTEXT_INTERFACE

Comment (pathakraul) [reply include/librpmi.h:381]: Either this or moving the group again to satisfy the symbol visibility is fine.

Comment (pathakraul) [reply include/librpmi.h:381]: ok
